utils/service_discovery/abstract_sd_backend.py

```diff
@@ -41,14 +41,25 @@ def _render_template(self, init_config_tpl, instance_tpl, variables):
             for key in tpl:
                 # iterate over template variables found in the templates
                 for var in self.PLACEHOLDER_REGEX.findall(str(tpl[key])):
-                    if var.strip('%') in variables and variables[var.strip('%')]:
+                    if var.strip('%') in variables and variables[var.strip('%')] is not None:
```

Comment:
You are stripping your `var` twice here. (And even a third time below).

```python
var_value = variables.get(var.strip('%'))
if var_value is not None:
   ...
````

is simpler to read and more efficient.